packages/block-editor/src/hooks/child-layout.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+
+function helpText( selfStretch ) {
+	switch ( selfStretch ) {
+		case 'fill':
+			return __( 'Stretch to fill available space.' );
+		case 'fixed':
+			return __( 'Specify a fixed width.' );
+		default:
+			return __( 'Fit contents.' );
+	}
+}
+
+/**
+ * Inspector control panel containing the child layout related configuration
+ *
+ * @param {Object} props                        Block props.
+ * @param {Object} props.attributes             Block attributes.
+ * @param {Object} props.setAttributes          Function to set block attributes.
+ * @param {Object} props.__unstableParentLayout
+ *
+ * @return {WPElement} child layout edit element.
+ */
+export function ChildLayoutEdit( {
+	attributes,
+	setAttributes,
+	__unstableParentLayout: parentLayout,
+} ) {
+	const { style = {} } = attributes;
+	const { layout: childLayout = {} } = style;
+	const { selfStretch, flexSize } = childLayout;
+
+	return (
+		<>
+			<ToggleGroupControl
+				label={ childLayoutOrientation( parentLayout ) }
+				value={ selfStretch || 'fit' }
+				help={ helpText( selfStretch ) }
+				onChange={ ( value ) => {
+					setAttributes( {
+						style: {
+							...style,
+							layout: {
+								...childLayout,
+								selfStretch: value,
+							},
+						},
+					} );
+				} }
```

Comment:
Could we also reset the `flexSize` value when switching away from the `Fixed` width option so that it isn't superfluously retained in the block attributes?